pulp_herminig/app/tasks/benchmark.py

```diff
@@ -1,15 +1,23 @@
 import logging
+import random
 import time
 
-from pulpcore.plugin.models import Task
-from pulpcore.plugin.tasking import dispatch
+logger = logging.getLogger(__name__)
 
-log = logging.getLogger(__name__)
+random.seed(0)  # Every time you start Pulp, make the tasking failure pattern randomly deterministic
 
 
-def noop():
-    pass
+def test_task(sleep_secs, failure_probability):
+    """
 
+    Args:
+        sleep_secs: The amount of time in seconds this task should sleep for.
+        failure_probability: The probability that the task will fail. This is expected to be in the
+            range [0.0, 1.0].
 
-def sleep(sec):
-    time.sleep(sec)
+    Returns: None
+
+    """
+    time.sleep(sleep_secs)
+    if failure_probability > random.random():
```

Comment:
I think, this is the wrong way around.
(i'd think, f_p = 0 means all tasks suceed)

Reply:
I think this is right. When I run `http POST http://pulp3-source-fedora34.localhost.example.com/pulp/api/v3/herminig/tasking_benchmark/` I get:

```
[
  {
    "pulp_href": "/pulp/api/v3/task-groups/065fc0f9-7602-4717-9d95-42df41f578bd/",
    "description": "Tasking system benchmark tasks",
    "all_tasks_dispatched": true,
    "waiting": 0,
    "skipped": 0,
    "running": 0,
    "completed": 4,
    "canceled": 0,
    "failed": 0,
    "group_progress_reports": []
  }
]
```

Reply:
I must have misinterpreted it.

For me, this is much easier to read (first measurement, then reference):
```suggestion
    if random.random() < failure_probability:
```